cmake/GinkgoConfig.cmake.in

```diff
@@ -207,4 +213,20 @@ if((NOT GINKGO_BUILD_SHARED_LIBS) AND GINKGO_HAVE_TAU)
     find_package(PerfStubs REQUIRED)
 endif()
 
+# Check that the same compilers as for Ginkgo are used
+function(_ginkgo_check_compiler lang)
+    if(NOT ${CMAKE_${lang}_COMPILER} STREQUAL ${GINKGO_${lang}_COMPILER})
+        set(_compiler_short "${CMAKE_${lang}_COMPILER_ID}:${CMAKE_${lang}_COMPILER_VERSION}")
```

Comment:
Do we need to use the compiler path here? We only use the ID to determine the rest anyways. And IIRC we propagate the CUDA host compiler, though that may fail if enable_language(CUDA) was called before (as I discovered when working on #1368)

Reply:
Yes, because `*_COMPILER_ID` and `*_COMPILER_VERSION` don't exist for the cuda host compiler. And we only set the cuda host compiler, if it is not already set in the downstream project.

Reply:
I think as CUDA is the only language that uses a separate host compiler, we could also omit this second if and check the ID directly, and put the check for the host compiler directly into line 229